### PR TITLE
Fix userstore domain incorrect response issue 

### PR DESCRIPTION
**Pull request by shashimalcse**
When user store domain is incorrect, it gives 
```{"schemas":["urn:ietf:params:scim:api:messages:2.0:Error"],"detail":"Cannot add user through scim to user store . SCIM is not enabled for user store <domain name>","status":"500"}```

This issue is fixed in this PR.

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 1]
```diff
@@ -2721,7 +2721,7 @@ private boolean isMemberAttributeRequired(Map<String, Boolean> requiredAttribute
     }
 
     @Override
-    public void deleteGroup(String groupId) throws NotFoundException, CharonException {
+    public void deleteGroup(String groupId) throws NotFoundException, CharonException, BadRequestException {
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:2724]: Changing the public method will lead to some issues with the exsiting usages. The best way I think we can create a separate method with the same impl and add the extra exception.
@senthalan any input on this.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 2]
```diff
@@ -6081,4 +6084,5 @@ private List<String> addDomainToNames(String userStoreDomainName, List<String> g
         return groupsList.stream().map(groupName -> UserCoreUtil.addDomainToName(groupName, userStoreDomainName))
                 .collect(Collectors.toList());
     }
+
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:6087]: Shall we remove this unnecessary extra line.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 1 above)
Comment (senthalan) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:2724]: yes. we can't change the signature of a public method. Can't we throw this as a CharonException with a specific error code and then process in the API layer. 

How we are handling the scim disabled userstores case

**Comment by shashimalcse** [comment]
Will be fixed with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/437

**Pull request closed without merging**